rest/handler.go

```diff
@@ -330,8 +330,12 @@ func (h *handler) validateAndWriteHeaders(method handlerMethod, accessPermission
 							return ErrInvalidLogin
 						}
 					}
-					base.InfofCtx(h.ctx(), base.KeyHTTP, "Error trying to get db %s: %v", base.MD(keyspaceDb), err)
-					return err
+					// look for db in config registry
```

Comment:
I'm assuming this is here so that we don't error here before we can get to the custom handling in deleteDB - is that correct?  Is this safe for all other calls?

Looking at the handling immediately below, for example - it looks like we're not going to perform the correct offline checks for a db-scoped request that's in this situation.  Further down, checkAuth will no-op if the dbContext is nil.  Where do we guard that non-deleteDB requests (a) don't eventually panic on a nil db ctx, (b) don't leak information via inconsistent error codes, or (c) succeed when they shouldn't.

Reply:
The answer is that I'm not sure about this.

Here's what I do know:
- jenkins integration tests continue to pass, but they largely don't use a bootstrap connection
- we almost always check `h.db == nil` on spot knowledge.

I think this is the right long term fix, but we don't have good test coverage for it.

We could limit this allowance in this handler whether to the request is admin and only allow this for deleting a db (and maybe upserting a config). We do check admin auth here https://github.com/couchbase/sync_gateway/blob/master/rest/handler.go#L310

Reply:
I fixed and added tests for this, I think.